### ISSUE TITLE
The condition at line 188 of ChunkedUploadProvider.java is never satisfied so the success callback is never called.

### ISSUE THREAD

**Issue by baywet** [microsoftgraph/msgraph-sdk-java#489]
The condition at line 188 of ChunkedUploadProvider.java is never satisfied so the success callback is never called.
 ```Java
          if (result.uploadCompleted()) {
                callback.progress(this.streamSize, this.streamSize);
                callback.success((UploadType) result.getItem());
                break;
            } 
```

_Originally posted by @RegisStGelais in https://github.com/microsoftgraph/msgraph-sdk-java/issues/472#issuecomment-696696095_

**Comment by baywet** [comment]
Hey @RegisStGelais,
After investigating it, I was able to replicate the issue. I created another issue for clarity.
I also authored #490 to fix this issue which will be close when the PR gets merged.

**Comment by ravisamani** [comment]
We also faced this issue only after upgrading to 2.2.0 
It was working on 1.7.0

@baywet any workaround suggestion until 2.3.0 is out?

**Comment by baywet** [comment]
Upgrade to 2.1.0 in the meantime?
No promises here, but as we only have one bug left in 2.3.0 we could potentially be looking at an earlier release date.
